signal_scanner_bot/transport.py

```diff
@@ -1,72 +1,108 @@
+
+    pass
+
+
+@Stream.event_stream
+class UserStream(peony.EventStream):
+    async def stream_request(self):
+        """
+        Basic stream set up
+        """
+        await self.api.client.stream.statuses.filter.post(
+            follow=",".join(env.TRUSTED_TWEETERS)
+        )
+
+    @peony.events.on_connected.handler
+    def connection(self):
+        """
+        Action to perform on connection
+        """
+        print("Connected to Twitter Stream API")
+
+    @peony.events.on_retweeted_status.handler
+    async def on_retweet(self, data):
+        """
+        Filter tweets from followed users on specified hashtags
+        Execute something (send to Signal or whatever)
+        """
+        pass
+
+    @peony.events.on_tweet.handler
+    async def on_tweet(self, data):
+        """
+        Filter tweets from followed users on specified hashtags
+        Execute something (send to Signal or whatever)
+        """
+        print("Got a tweet\n\n\n\n\n")
+        if _filter_hashtags(data, env.RECEIVE_HASHTAGS):
+            pprint.pprint(data)
+
+    @peony.events.reconnecting_in.handler
+    async def reconnecting(self, data):
+        """
+        Action to perform on reconnect
+        """
+        print("reconnecting in %ss" % data.reconnecting_in)
+
+    @peony.events.on_restart.handler
+    async def restart_notice(self):
+        """
+        Action to perform on reconnect
+        """
+        print("*Stream restarted*\n" + "-" * 10)
+        await self.api.client.stream.statuses.filter.post(
+            follow=",".join(env.TRUSTED_TWEETERS)
+        )
+
+    @peony.events.on_dm.handler
+    async def direct_message(self, data):
+        pass
+
+    @peony.events.friends.handler
+    async def pass_friends(self):
+        pass
+
+    @peony.events.default.handler
+    async def default(self, data):
+        pass
 
 
 async def twitter_to_queue():
-    """
-    Asynchronous interface for the twitter-to-signal loop.
-    """
-    urllib3_error_message = (
-        "There was a Twitter API connection error, restarting Twitter API stream"
-    )
-    loop = asyncio.get_event_loop()
-    with concurrent.futures.ThreadPoolExecutor() as pool:
-        try:
-            await loop.run_in_executor(pool, _twitter_to_queue)
-        except urllib3.exceptions.ProtocolError as err:
-            log.warning(urllib3_error_message + ": Protocol Error\n" + str(err))
-        except urllib3.exceptions.ReadTimeoutError as err:
-            log.warning(urllib3_error_message + ": Read Timeout Error\n" + str(err))
-        except Exception as err:
-            log.error(
-                "Exception occurred in the Twitter to Queue pipeline, halting process"
-            )
-            log.exception(err)
-            signal.panic(err)
-            env.STATE.STOP_REQUESTED = True
-            raise
+    log.info("Starting Twitter Event Stream")
+    stream = peony.PeonyClient(**env.API_KEYS)
+    stream.run()
```

Comment:
Something is wrong here and I'm not sure what happened. I thought I had this set to:

```
log.inf("Starting Twitter Event Stream")
stream = Stream(**env.API_KEYS)
stream.run()
```

Reply:
Ok so I was actually able to kind of get this working using the following:

```
async def twitter_to_queue():
    log.info("Starting Twitter Event Stream")
    stream_obj = env.CLIENT.api.client.stream.statuses.filter.post(
        follow=",".join(env.TRUSTED_TWEETERS)
    )
    async with stream_obj as stream:
        async for data in stream:
            if events.on_connect(data):
                print("Connected to the stream")
            elif events.on_tweet(data):
                pprint.pprint(data)
```

But that still leaves me pretty confused about how to do this in a more consumable manner. I'm ok with doing it this way I guess, but it feels like it isn't really the "right" way to do it.